pkg/vms/actions.es6

```diff
@@ -0,0 +1,179 @@
+import cockpit from 'base1/cockpit';
+import Libvirt from 'vms/libvirt';
+import { getRefreshInterval } from 'vms/selectors';
+import { VMS_CONFIG } from "vms/config";
+
+/**
+ * All actions dispatchableby in the application
+ */
+
+// --- Provider actions -----------------------------------------
+export function getAllVms () {
+  return virt('GET_ALL_VMS');
+}
+
+export function getVm (lookupId) {
+  return virt('GET_VM', { lookupId }); // provider-specific (i.e. libvirt uses vm_name)
+}
+
+export function shutdownVm (name) {
+  return virt('SHUTDOWN_VM', { name });
+}
+
+export function forceVmOff (name) {
+  return virt('FORCEOFF_VM', { name });
+}
+
+export function rebootVm (name) {
+  return virt('REBOOT_VM', { name });
+}
+
+export function forceRebootVm (name) {
+  return virt('FORCEREBOOT_VM', { name });
+}
+
+export function startVm (name) {
+  return virt('START_VM', { name });
+}
+
+/**
+ * Helper for dispatching virt provider methods.
+ *
+ * Lazily initializes the virt provider and dispatches given method on it.
+ */
+function virt (method, action) {
+  return (dispatch, getState) => getVirtProvider({ dispatch, getState }).then(provider => {
+    if (method in provider) {
+      console.log(`Calling ${provider.name}.${method}(${JSON.stringify(action)})`);
+      return dispatch(provider[method](action));
+    } else {
+      console.warn(`method: '${method}' is not supported by provider: '${provider.name}'`);
+    }
+  }).catch(err => {
+    console.error('could not detect any virt provider');
+  })
+}
+
+function getVirtProvider(store) {
+  const state = store.getState();
+  if (state.config.provider) {
+    return cockpit.resolve(state.config.provider);
+  } else {
+    const deferred = cockpit.defer();
+    console.log('Discovering provider');
```

Comment:
Good point. Recently it's considered to be exclusive either/or.

So far I'm not sure if merging data from different providers will be needed or not. 
I think not, but I'm not sure.

If we keep all the providers together, we get possibility to switch them at runtime. It might be useful for testing of our code same as the providers itself. For sure there will be different dataset retrieved from different providers, the administrator might be interested in seeing the difference.

The overhead is small - one JS file per provider as the provider API is designed.

I need to think more about it.